### PR TITLE
slight rewrite of explanation of _ underscore protection

### LINKED ISSUE #59: DNSDIR:  Clarify DNSsec usage
I'd prefer some parts around DNSSEC better clarified.  The 5.2 integrity
section mentions (twice) a compromise of servers that *host* a zone.  I
consider "host" a misleading word here, as the important part for integrity is
only the server that signs the zone.  (servers that serve the zones would be
more about 5.3 availability)

https://datatracker.ietf.org/doc/review-ietf-dance-architecture-06-dnsdir-early-cunat-2024-07-19/

### PR DESCRIPTION
**Pull request by mcr**
close #59

### CODE REVIEW

File: draft-ietf-dance-architecture.md [hunk 1]
```diff
@@ -418,9 +418,13 @@ Compartmentalizing failure domains within an application is a well-known archite
 Within the context of protecting DNS-based identities, this compartmentalization may manifest by hosting an identity zone on a DNS server which only supports the resource record types essential for representing device identities.
 This can prevent a compromised identity zone DNS server from presenting records essential for impersonating web sites under the organization’s domain name.
 
-The naming pattern suggested in <https://datatracker.ietf.org/doc/html/draft-huque-dane-client-cert> includes
-an underscore label (_device) which also prevents the issuance of Web PKI-validating certificates in the
-event a DNS server hosting a client identity zone, which is capable of presenting A and AAAA records, is compromised.
+The naming pattern suggested in {{I-D.ietf-dane-client-certs}} includes
+an underscore label (_device).
+The underscore is not a valid character for in names used in the Web PKI.
```
Comment (oej) [inline draft-ietf-dance-architecture.md:423]: "for in names" - maybe remove the "in"

**Review by oej — CHANGES_REQUESTED** [review]

File: draft-ietf-dance-architecture.md (hunk 1 above)
Comment (mcr) [inline draft-ietf-dance-architecture.md:423]:
```suggestion
The underscore is not a valid character for names used in the Web PKI.
```

**Review by oej — APPROVED** [review]
ok

File: draft-ietf-dance-architecture.md [hunk 2]
```diff
@@ -418,9 +418,13 @@ Compartmentalizing failure domains within an application is a well-known archite
 Within the context of protecting DNS-based identities, this compartmentalization may manifest by hosting an identity zone on a DNS server which only supports the resource record types essential for representing device identities.
 This can prevent a compromised identity zone DNS server from presenting records essential for impersonating web sites under the organization’s domain name.
 
-The naming pattern suggested in <https://datatracker.ietf.org/doc/html/draft-huque-dane-client-cert> includes
-an underscore label (_device) which also prevents the issuance of Web PKI-validating certificates in the
-event a DNS server hosting a client identity zone, which is capable of presenting A and AAAA records, is compromised.
+The naming pattern suggested in {{I-D.ietf-dane-client-certs}} includes
```
Comment (mcr) [inline draft-ietf-dance-architecture.md:421]:
```suggestion
The naming pattern suggested in {{I-D.ietf-dane-client-auth}} includes
```

**Pull request merged**